File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(), skill.end());
        int sum=skill[0]+skill[skill.size()-1];
        int sz = skill.size()-1;
        long long ans = skill[0]*skill[skill.size()-1];
        for(int i = 1; i< skill.size()/2; i++)
        {
            if(skill[i] + skill[sz-i] != sum)
                return -1;
            else
                ans+=(skill[i]*skill[sz-i]);
        }
        return ans;
        
    }
};

File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cs
public class Solution {
    public long DividePlayers(int[] skill) {

        Array.Sort(skill);
        int sz = skill.Length-1;
        int sum=skill[0]+skill[sz];
        
        for(int i = 1; i< skill.Length/2; i++)
        {
            if(skill[i] + skill[sz-i] != sum)
                return -1;
        }
        long ans =0;
        for(int i  =0; i< skill.Length/2; i++)
        {
            ans+=(skill[i]*skill[sz-i]);
        }return ans;
        
    }
}